src/dsp_tools/commands/xmlupload/models/serialise/serialise_value.py

```diff
@@ -25,27 +25,45 @@ class SerialiseValue(Protocol):
     def serialise(self) -> dict[str, Any]:
         """Serialise the value."""
 
+    def _get_optionals(self) -> dict[str, str]:
+        optionals = {"knora-api:valueHasComment": self.comment} if self.comment else {}
+        if self.permissions:
+            optionals["knora-api:hasPermissions"] = self.permissions
+        return optionals
+
```

Comment:
Here you mix up composition (protocols) and inheritance (abstract base classes). 

If you prefer composition (protocols), then you shouldn't implement anything on the protocol level. All implementation must go into the child classes. Protocols are pure structure (skeletons), no flesh.

If you prefer inheritance (abstract base classes), then you can leave your code as it is, but annotate it differently:
- SerialiseValue becomes an ABC (https://docs.python.org/3/library/abc.html#abc.ABC)
- SerialiseValue.serialise() should be decorated as abstract method (https://docs.python.org/3/library/abc.html#abc.abstractmethod). By doing so, you enforce child classes to implement it.

In your case, I would go for ABCs, because it's tiresome to repeat the implementation of _get_optionals() all over the place.